### DISCUSSION TITLE
BUG :: Product recommendations like recently viewed which required local storage write not working while works with Luma as expected

### DISCUSSION THREAD

**Discussion by sandipansaha** [breezefront/community#298] [1. Help] [answered]
Hi @vovayatsyuk 

We have found a very interesting bug.

When we use a product recommendations configured with Recently Viewed rule that requires local storage data like `window.localStorage.getItem("ds-view-history-time-decay")`, it gives error while the local storage variable is not set and not able to write it.

Error:
```javascript
index.js:7 Uncaught TypeError: Cannot read properties of null (reading 'pop')
    at Z (index.js:7:12697)
    at t.value (index.js:7:18689)
    at Magento_ProductRecommendationsLayout/js/recsFetcher (recsFetcher.min.js:6:8)
    at HTMLBodyElement.<anonymous> (main.min.js:35:327)
    at d (cash-8.1.5.min.js:1:2014)
    at u.each (cash-8.1.5.min.js:1:2090)
    at HTMLDocument.<anonymous> (main.min.js:34:27)
    at HTMLDocument.et (cash-8.1.5.min.js:1:16182)
    at HTMLDocument.<anonymous> (cash-8.1.5.min.js:1:14409)
    at d (cash-8.1.5.min.js:1:2014)
``` 
![image](https://github.com/breezefront/community/assets/4614581/010ea16f-8e48-4e9f-940c-1c900648cd28)

Please follow the following process to replicate the issue -

1. Open a browser window like Google chrome and clear the local storage for the URL - 
2. Open the Product Page URL -
3. Check console for the error.
4. Reload the page and notice the same error persists.

Now with LUMA, follow the same steps with the URL -  or use ?breeze=0 with the breeze product URL, you can notice the error does not occur on page reload as with LUMA it is able to re-write the required local storage data.

While doing the debug, I have found that the error occurs in https://magento-recs-sdk.adobe.net/v2/index.js and the re-write of the local storage variable happens in https://commerce.adobedtm.com/v6/ds.min.js

![image](https://github.com/breezefront/community/assets/4614581/4f165194-219b-4a19-84a8-5d07ade77b5c)
![image](https://github.com/breezefront/community/assets/4614581/187fc571-3973-4d7e-a459-d3aaaae399b3)

With debugger, I have noticed that while running in Breeze the function in `ds.min.js` that is responsible for re-write, only gets initialised but does not get executed.
But to my surprise it works with LUMA.

The `dataServicesBase` file is being configured in breeze and it is configured above product recommendations.
In my local I have tried to remove the `load` section to exclude the `dataServicesBase` from loading dynamically, but it still being loaded dynamically. Not sure why? Any reason for that behaviour?

![image](https://github.com/breezefront/community/assets/4614581/0b4f5489-ce06-4b12-917e-e3c99de70dd6)
![image](https://github.com/breezefront/community/assets/4614581/ebb66635-390c-45f3-8245-f93b2c69d89d)

Hence, I was wondering if you could help/guide me on the correct path. Is it something due to dynamic JS or with local storage or I am doing something wrong.
If you need any further information, I am available any time to discuss this with you. I appreciate your help very much.

Thanks
Sandipan

**Comment by vovayatsyuk** [comment, marked as answer]
Thanks!

This commit should fix it: https://github.com/breezefront/module-breeze-magento-enterprise/commit/f027a6032e44fe20b4291c233d0d8f7688b28cfb

**Reply by sandipansaha** [reply]
Thank @vovayatsyuk , it fixed the issue. Please let me know once published.

**Reply by vovayatsyuk** [reply]
Done!
